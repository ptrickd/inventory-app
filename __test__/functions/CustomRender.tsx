//Testing library
import { render, RenderOptions } from "@testing-library/react";

//Mocked Files
import { mockedRouter } from "./mockedRouter";
// import { RouterContext } from "next/dist/shared/lib/router-context";
import { RouterContext } from "next/dist/shared/lib/router-context.shared-runtime";
//Context
import { UserContext } from "../../contexts/UserContext";
import { ReportsContext } from "../../contexts/ReportsContext";
import { CategoriesContext } from "../../contexts/CategoriesContext";
import { ProductsContext } from "../../contexts/ProductsContext";

//ContextProps
import {
  CategoriesContextProps,
  ProductsContextProps,
  UserContextProps,
  ReportsContextProps,
} from "./mockedContexts";

const AllTheProviders: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  return (
    <RouterContext.Provider value={mockedRouter({})}>
      <UserContext.Provider value={UserContextProps}>
        <ReportsContext.Provider value={ReportsContextProps}>
          <CategoriesContext.Provider value={CategoriesContextProps}>
            <ProductsContext.Provider value={ProductsContextProps}>
              {children}
            </ProductsContext.Provider>
          </CategoriesContext.Provider>
        </ReportsContext.Provider>
      </UserContext.Provider>
    </RouterContext.Provider>
  );
};

const CustomRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export default CustomRender;
