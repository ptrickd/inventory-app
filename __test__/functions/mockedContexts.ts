/*
put constant here
*/
export const CURRENT_USER_ID = "789456";
export const CURRENT_USER_EMAIL = "test@email.com";

export const REPORT_ID = "789456";
export const CARROT_PRODUCT_ID = "123456";

export const PRODUCE_CATEGORY_ID = "123456";
export const COOLER_CATEGORY_ID = "123457";

export const UserContextProps = {
  currentUser: { id: CURRENT_USER_ID, email: CURRENT_USER_EMAIL },
  setCurrentUser: jest.fn(),
  loggedIn: true,
  setLoggedIn: jest.fn(),
  setToken: jest.fn(),
  logout: jest.fn(),
};

export const ReportsContextProps = {
  hasReport: true,
  reports: [
    {
      id: REPORT_ID,
      userId: CURRENT_USER_ID,
      dateEndingCycle: new Date("2023-04-06T00:00:00.000Z"),

      products: [
        {
          productId: CARROT_PRODUCT_ID,
          amount: 6,
          name: "Carrots",
          categoryId: PRODUCE_CATEGORY_ID,
        },
      ],
      hasBeenSubmitted: false,
      dateCreated: new Date("2023-04-06T00:00:00.000Z"),
      dateSubmitted: new Date("2023-04-06T00:00:00.000Z"),
    },
  ],
  createNewReport: jest.fn(),
  deleteLocalReport: jest.fn(),
};

export const CategoriesContextProps = {
  hasCategory: true,
  setCategories: jest.fn(),
  categories: [
    {
      id: PRODUCE_CATEGORY_ID,
      name: "Produce",
    },
    {
      id: COOLER_CATEGORY_ID,
      name: "Cooler",
    },
  ],
  createCategoryApi: jest.fn(),
  deleteCategoryApi: jest.fn(() => {
    error: true;
  }),
};

export const ProductsContextProps = {
  hasProduct: true,
  products: [
    {
      id: CARROT_PRODUCT_ID,
      categories: [
        {
          categoryId: PRODUCE_CATEGORY_ID,
          currentAmount: 6,
          previousAmount: 5,
          position: 0,
        },
      ],
      name: "Carrots",
      unit: "ea",
    },
  ],
  productsByCategory: [
    {
      id: CARROT_PRODUCT_ID,
      categoryId: PRODUCE_CATEGORY_ID,
      currentAmount: 6,
      previousAmount: 5,
      name: "Carrots",
      unit: "ea",
      position: 0,
    },
  ],
  updateProducts: jest.fn(),
  setCategoryId: jest.fn(),
  addProduct: jest.fn(),
  deleteProductApi: jest.fn(),
  editProductApi: jest.fn(),
};
