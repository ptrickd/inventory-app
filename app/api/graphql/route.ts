//NextJs
import { NextApiRequest } from "next";
import { NextRequest } from "next/server";

//GraphQL
import { ApolloServer } from "@apollo/server";
import { schema } from "../../../graphql/schema";
import { startServerAndCreateNextHandler } from "@as-integrations/next";

//Models
import dbConnect from "../../../utils/dbConnect";

//Auth
import jwt from "jsonwebtoken";

dbConnect();

const getUser = (req: NextRequest) => {
  try {
    console.log(`req: ->`);
    console.log(req.headers.get("authorization"));
    if (req) {
      const tokenWithBearer = req.headers.get("authorization") || "";
      const token = tokenWithBearer.split(" ")[1];

      return jwt.verify(token, process.env.RESTO_JWT_SECRET as string);
    }
    return null;
  } catch (err) {
    return null;
  }
};

const apolloServer = new ApolloServer({ schema });

const handler = startServerAndCreateNextHandler(apolloServer, {
  context: async (req: NextRequest, res) => ({
    req,
    res,
    csrfPrevention: true, //cors
    user: getUser(req),
  }),
});
export { handler as GET, handler as POST };
