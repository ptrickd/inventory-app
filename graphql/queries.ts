//GraphQL
import { gql } from "@apollo/client";

//Authentication
export const REGISTER = gql`
  mutation Register($email: String!, $password: String!) {
    register(email: $email, password: $password) {
      user {
        id
        email
        theme
      }
      error
    }
  }
`;
export const LOGIN = gql`
  mutation Login($email: String!, $password: String!) {
    login(email: $email, password: $password) {
      token
      user {
        id
        email
        theme
      }
      error
    }
  }
`;

export const GET_CURRENT_USER = gql`
  query CurrentUser {
    currentUser {
      id
      email
      theme
    }
  }
`;

//Product
export const GET_PRODUCTS_BY_CATEGORY = gql`
  query ProductsByCategory($categoryId: String) {
    productsByCategory(categoryId: $categoryId) {
      id
      name
      currentAmount
      previousAmount
      categoryId
      unit
    }
  }
`;
export const CREATE_PRODUCT = gql`
  mutation CreateProduct($name: String!, $categoryId: String!, $unit: String!) {
    createProduct(name: $name, categoryId: $categoryId, unit: $unit) {
      id
      name
      categories {
        categoryId
        currentAmount
        previousAmount
      }
      unit
      error
    }
  }
`;
export const DELETE_PRODUCT = gql`
  mutation DeleteProduct($productId: ID!) {
    deleteProduct(productId: $productId) {
      id
      name
    }
  }
`;
export const EDIT_PRODUCT = gql`
  mutation EditProduct(
    $productId: ID
    $name: String
    $categoryId: String
    $unit: String
  ) {
    editProduct(
      productId: $productId
      name: $name
      categoryId: $categoryId
      unit: $unit
    ) {
      id
      name
      categoryId
      unit
    }
  }
`;

//Category
export const GET_CATEGORIES = gql`
  query GetCategories {
    categories {
      id
      name
      error
    }
  }
`;

export const GET_NUM_CATEGORIES = gql`
  query GetNumOfCategories {
    numOfCategories
  }
`;

export const CREATE_CATEGORY = gql`
  mutation CreateCategory($name: String!) {
    createCategory(name: $name) {
      id
      name
      error
    }
  }
`;

export const GET_CATEGORY = gql`
  query GetCategory($categoryId: ID!) {
    category(categoryId: $categoryId) {
      id
      name
    }
  }
`;
