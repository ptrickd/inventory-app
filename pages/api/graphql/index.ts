
//GraphQL
import { ApolloServer, gql } from 'apollo-server-micro'

//Controller
import {
    createProduct,
    getProducts,
    getProductsByCategory,
    editProduct,
    deleteProduct
} from '../../../controllers/product.controller'
import {
    getCategories,
    createCategory,
    getCategory,
    editCategory,
    deleteCategory
} from '../../../controllers/category.controller'

const typeDefs = gql`
    type Product {
        id: ID
        name: String
        amount: Int
        categoryId: ID
    }

    type Category {
        id: ID
        name: String
    }

    type Query {
        getProducts: [Product]
        getProductsByCategory(categoryId: String): [Product]

        getCategory(categoryId: ID):Category
        getCategories: [Category]
    }

    type Mutation {
        createProduct(name:String, amount:Int, categoryId: String): Product
        editProduct(id:ID, name:String, categoryId: String): Product
        deleteProduct(productId: ID): Product
        
        createCategory(name:String): Category
        editCategory(categoryId:ID, name:String): Category
        deleteCategory(categoryId:ID): Category
    }
`

interface IProduct {
    id: string
    name: string
    amount: number
    categoryId: string
}

interface ICreateProduct {
    name: string
    amount: number
    categoryId: string
}

interface IIds {
    [propName: string]: string
}

interface IEditProduct {
    id: string
    name: string
    categoryId: string
}

interface ICategory {
    categoryId: string
    name: string
}


const resolvers = {
    Query: {
        getProducts: async () => {
            try {
                const products = await getProducts()
                return products.map(({ id, amount, name, categoryId }: IProduct) => ({
                    id,
                    name,
                    amount,
                    categoryId
                }))
            } catch (err) {
                console.log(err)
            }
        },
        getProductsByCategory: async (_: any, { categoryId }: IIds) => {
            try {
                const products = await getProductsByCategory(categoryId)
                if (!products) throw new Error('No products found')
                return products.map(({ id, amount, name, categoryId }: IProduct) => ({
                    id,
                    name,
                    amount,
                    categoryId
                }))
            }
            catch (err) {
                console.log(err)
            }
        },
        getCategory: async (_: any, { categoryId }: IIds) => {
            try {
                const category = await getCategory(categoryId)
                if (!category) throw new Error("No Category Found");

                return category
            }
            catch (err) {
                console.log(err)
            }
        },
        getCategories: async () => {
            try {
                console.log('in getCategories')
                const categories = await getCategories()
                if (!categories) throw new Error("No Categories Found")
                return categories.map(({ id, name }) => ({
                    id, name
                }))
            }
            catch (err) {
                console.log(err)
            }
        }
    },
    Mutation: {
        createProduct: async (_: any, { name, amount, categoryId }: ICreateProduct) => {

            let product = await createProduct(name, amount, categoryId)
            return product
        },
        editProduct: async (_: any, { id, name, categoryId }: IEditProduct) => {
            let product = await editProduct(id, name, categoryId)
            if (!product) throw new Error('No product found')
            return product
        },
        deleteProduct: async (_: any, { productId }: IIds) => {
            let product = await deleteProduct(productId)
            if (!product) throw new Error('No product found')
            return product
        },
        createCategory: async (_: any, { name }: IIds) => {
            console.log('createCategory', name)
            let category = await createCategory(name)
            if (!category) throw new Error("No Category Create")
            return category

        },
        editCategory: async (_: any, { categoryId, name }: ICategory) => {
            let editedCategory = await editCategory(categoryId, name)
            if (!editedCategory) throw new Error("No Category Found")
            return editedCategory
        },
        deleteCategory: async (_: any, { categoryId }: IIds) => {
            let deletedCategory = await deleteCategory(categoryId)
            if (!deletedCategory) throw new Error("No Category Found")
            return deletedCategory
        }
    }
}


const apolloServer = new ApolloServer({ typeDefs, resolvers })

export const config = {
    api: {
        bodyParser: false
    }
}

export default apolloServer.createHandler({ path: "/api/graphql" })
