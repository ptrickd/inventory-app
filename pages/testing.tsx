//React
import React, { useState, useEffect, useContext } from "react";

//Context
import { UserContext } from "../contexts/UserContext";

//Material UI
import { styled } from "@mui/material-pigment-css";
import Container from "@mui/material-pigment-css/Container";
import Paper from "@mui/material/Paper";
import Box from "@mui/material-pigment-css/Box";
import Typography from "@mui/material/Typography";
import IconButton from "@mui/material/IconButton";
import Grid from "@mui/material-pigment-css/Grid";

//Icons
import ArrowBackIosIcon from "@mui/icons-material/ArrowBackIos";
import ArrowForwardIosIcon from "@mui/icons-material/ArrowForwardIos";

//Component
import FirstReport from "../components/FirstReport";
import FirstCategory from "../components/FirstCategory";
import FirstProduct from "../components/FirstProduct";

const PREFIX = "Test";

const classes = {
  root: `${PREFIX}-root`,
  styledPaper: `${PREFIX}-styled-paper`,
};

const Root = styled(Container)(() => ({
  [`&.${classes.root}`]: {
    width: "100%",
    minHeight: "calc(100vh - 120px)",
    margin: 5,
    // display: "flex",
    // flexDirection: "column",
    // justifyContent: "center",
    // background: "red",
  },
}));

const StyledPaper = styled(Paper)(() => ({
  [`&.${classes.styledPaper}`]: {
    width: "100%",
    minHeight: "70vh",
    display: "flex",
    flexDirection: "row",
    justifyContent: "center",
  },
}));

const Test: React.FC = () => {
  //Context
  const { currentUser } = useContext(UserContext);

  //useState
  const [pageNumber, setPageNumber] = useState(0);

  //Functions
  const componentToRender = () => {
    switch (pageNumber) {
      case 0:
        return <FirstReport />;
        break;
      case 1:
        return <FirstCategory />;
        break;

      case 2:
        return <FirstProduct />;
        break;
      default:
        break;
    }
  };
  const decreasePageNumber = () => {
    if (pageNumber > 0) {
      setPageNumber(pageNumber - 1);
    }
  };

  const increasePageNumber = () => {
    if (pageNumber < 2) {
      setPageNumber(pageNumber + 1);
    }
  };

  //useEffect
  useEffect(() => {
    console.log(pageNumber);
  }, [pageNumber]);
  if (typeof currentUser === "undefined") return <div>Error</div>;
  return (
    <Root className={classes.root}>
      <StyledPaper className={classes.styledPaper} elevation={1}>
        <Box
          component="div"
          sx={{
            flexGrow: 1,
            // justifyContent: "center",
            // aligItems: "center",
            // alignContent: "center",
            // height: "100%",
          }}
        >
          <Grid
            container
            spacing={2}
            sx={{
              flexGrow: 1,
              justifyContent: "center",
              aligItems: "baseline",
              alignContent: "center",
              height: "100%",
            }}
          >
            <Grid item xs={1} sx={{ alignSelf: "center" }}>
              <IconButton onClick={decreasePageNumber}>
                <ArrowBackIosIcon />
              </IconButton>
            </Grid>

            <Grid item xs={8}>
              <Box
                component="div"
                // sx={{ flexGrow: 1, height: "100%", border: "1px solid" }}
              >
                {componentToRender()}
              </Box>
            </Grid>
            <Grid item xs={1} sx={{ alignSelf: "center" }}>
              <IconButton onClick={increasePageNumber}>
                <ArrowForwardIosIcon />
              </IconButton>
            </Grid>
            <Typography>Check this box to not see again!</Typography>
          </Grid>
        </Box>
      </StyledPaper>
    </Root>
  );
};

export default Test;
