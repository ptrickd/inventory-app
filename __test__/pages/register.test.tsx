//Testing
import { render, screen, waitFor, act } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

//Mocked structure
import { mockedRouter } from "../functions/mockedRouter";
import { MockedProvider } from "@apollo/client/testing";

//Mocked router
const useRouter = jest.spyOn(require("next/router"), "useRouter");

//Component
import Register from "../../pages/register";
import { RouterContext } from "next/dist/shared/lib/router-context.shared-runtime";

//Query
import { REGISTER } from "../../graphql/queries";

const mocks: any = [
  {
    request: {
      query: REGISTER,
      variables: {
        email: "test@email.com",
        password: "securepassword11",
      },
    },
    result: {
      data: {
        register: {
          token: "789456",
          user: {
            id: "123456",
            email: "test@email.com",
            theme: "dark",
          },
          error: null,
        },
      },
    },
  },
];

describe("<Register />", () => {
  it("render as intended", async () => {
    render(
      <RouterContext.Provider value={mockedRouter({})}>
        <MockedProvider mocks={mocks} addTypename={false}>
          <Register />
        </MockedProvider>
      </RouterContext.Provider>
    );

    //need  an heading
    const loginHeading = screen.getByRole("heading", { name: "Register" });
    expect(loginHeading).toBeVisible();

    //need an email input
    const emailInput = screen.getByRole("textbox", { name: "Email" });
    expect(emailInput).toBeVisible();

    //need an password input
    const PasswordInput = screen.getByLabelText(/password/i);
    expect(PasswordInput).toBeInTheDocument();

    //need a register button
    const registerButton = screen.getByRole("button", { name: "Register" });
    expect(registerButton).toBeVisible();
  });

  it("send data to register if values are valid", async () => {
    //mock router
    const router = { push: jest.fn() };
    useRouter.mockReturnValue(router);

    render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Register />
      </MockedProvider>
    );

    const user = userEvent.setup();

    //fill the email input with a valid email
    const EmailInput = screen.getByRole("textbox", { name: /email/i });
    await act(async () => {
      await user.click(EmailInput);
      await user.keyboard("test@email.com");
    });
    //fill the password input with a valid password
    const PasswordInput = screen.getByLabelText(/password/i);
    await act(async () => {
      await user.click(PasswordInput);
      await user.keyboard("securepassword11");
    });

    //click on the button register
    const RegisterButton = screen.getByRole("button", { name: /register/i });

    await user.click(RegisterButton);

    expect(router.push).toHaveBeenCalledWith("/login");
  });

  it("match the snapshot", async () => {
    const tree = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Register />
      </MockedProvider>
    );

    expect(tree).toMatchSnapshot();
  });
});
