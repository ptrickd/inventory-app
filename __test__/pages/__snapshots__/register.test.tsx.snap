// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style=""
  >
    <div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthXs Register-root css-16i7vig-MuiContainer-root"
      >
        <div
          class="Register-main MuiBox-root css-16nem1j"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter Register-title css-1s4drii-MuiTypography-root"
            style="--Typography-textAlign: center;"
          >
            Register
          </h2>
          <div
            class="css-skvc7k"
          >
            <form
              class="AuthForm-form"
            >
              <div
                class="MuiFormControl-root MuiTextField-root AuthForm-input css-1xp5r68-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for=":ra:"
                  id=":ra:-label"
                >
                  Email
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="off"
                    class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":ra:"
                    name="email"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-81qg8w"
                    >
                      <span>
                        Email
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiTextField-root AuthForm-input css-1xp5r68-MuiFormControl-root-MuiTextField-root"
                data-testid="password"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="password"
                  id="password-label"
                >
                  Password
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="off"
                    class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                    id="password"
                    input="password"
                    name="password"
                    type="password"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-81qg8w"
                    >
                      <span>
                        Password
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary AuthForm-button css-1abvrni-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="submit"
              >
                Register
              </button>
            </form>
          </div>
        </div>
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg Footer-root css-1jfhocz-MuiContainer-root"
        >
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
          />
          <div
            class="Footer-contact MuiBox-root css-0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
            >
              Contact
            </p>
            <div
              class="MuiBox-root css-0"
            >
              <a
                aria-label="github"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorIcon MuiIconButton-sizeMedium css-aixiz9-MuiButtonBase-root-MuiIconButton-root"
                data-mui-internal-clone-element="true"
                href="https://github.com/ptrickd/inventory-app"
                tabindex="0"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
                  data-testid="GitHubIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
                  />
                </svg>
              </a>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <a
                aria-label="email"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorIcon MuiIconButton-sizeMedium css-aixiz9-MuiButtonBase-root-MuiIconButton-root"
                data-mui-internal-clone-element="true"
                href="mailto:test@example.com"
                tabindex="0"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
                  data-testid="EmailIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2m0 4-8 5-8-5V6l8 5 8-5z"
                  />
                </svg>
              </a>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthXs Register-root css-16i7vig-MuiContainer-root"
    >
      <div
        class="Register-main MuiBox-root css-16nem1j"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter Register-title css-1s4drii-MuiTypography-root"
          style="--Typography-textAlign: center;"
        >
          Register
        </h2>
        <div
          class="css-skvc7k"
        >
          <form
            class="AuthForm-form"
          >
            <div
              class="MuiFormControl-root MuiTextField-root AuthForm-input css-1xp5r68-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":ra:"
                id=":ra:-label"
              >
                Email
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="off"
                  class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":ra:"
                  name="email"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-81qg8w"
                  >
                    <span>
                      Email
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root AuthForm-input css-1xp5r68-MuiFormControl-root-MuiTextField-root"
              data-testid="password"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="password"
                id="password-label"
              >
                Password
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="off"
                  class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                  id="password"
                  input="password"
                  name="password"
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-81qg8w"
                  >
                    <span>
                      Password
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary AuthForm-button css-1abvrni-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              Register
            </button>
          </form>
        </div>
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg Footer-root css-1jfhocz-MuiContainer-root"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
        />
        <div
          class="Footer-contact MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
          >
            Contact
          </p>
          <div
            class="MuiBox-root css-0"
          >
            <a
              aria-label="github"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorIcon MuiIconButton-sizeMedium css-aixiz9-MuiButtonBase-root-MuiIconButton-root"
              data-mui-internal-clone-element="true"
              href="https://github.com/ptrickd/inventory-app"
              tabindex="0"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
                data-testid="GitHubIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
                />
              </svg>
            </a>
          </div>
          <div
            class="MuiBox-root css-0"
          >
            <a
              aria-label="email"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorIcon MuiIconButton-sizeMedium css-aixiz9-MuiButtonBase-root-MuiIconButton-root"
              data-mui-internal-clone-element="true"
              href="mailto:test@example.com"
              tabindex="0"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
                data-testid="EmailIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2m0 4-8 5-8-5V6l8 5 8-5z"
                />
              </svg>
            </a>
          </div>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
